### ISSUE TITLE
Cannot use finetuning-scheduler with lightning==2.6.0

### ISSUE THREAD

**Issue by saydemr** [speediedan/finetuning-scheduler#25]
## 🐛 Bug

Hi,

I think there is an issue with the dependencies of the `finetuning-scheduler` in version 2.9.1. I cannot add `lightning==2.6.0` dependency with this version.

<img width="1342" height="155" alt="Image" src="https://github.com/user-attachments/assets/f3ad0388-a20c-49a7-a59d-6151e08b150f" />

On the other hand, when I do `uv add pytorch-lightning` it correctly adds `pytorch-lightning==2.6.0`

### To Reproduce

```
uv init ftsched
cd ftsched
uv add finetuning-scheduler
uv add lightning
uv pip list | grep lightning
```

### Expected behavior

uv adds `lightning==2.6.0`

### Environment

* CUDA:
        - GPU:
                - NVIDIA GeForce RTX 3070 Laptop GPU
        - available:         True
        - version:           12.8
* Packages:
        - finetuning-scheduler: 2.9.1
        - numpy:             2.3.5
        - pyTorch_debug:     False
        - pyTorch_git_version: 5811a8d7da873dd699ff6687092c225caffcf1bb
        - pyTorch_version:   2.9.1+cu128
        - pytorch-lightning: 2.5.6
        - tqdm:              4.67.1
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.13.7
        - version:           #1 SMP PREEMPT_DYNAMIC Thu Jun  5 18:30:46 UTC 2025

**Comment by speediedan** [comment]
Thanks for the prompt @saydemr! Have had lots on my plate and was a bit delayed validating the latest lightning patch/minor releases. I just published a patch release (2.9.2) that validates Lightning 2.6.0 support.

Note, as of FTS 2.9.x our versioning aligns with PyTorch minor releases rather than Lightning minor releases for a variety of reasons, see the [versioning documentation for more](https://finetuning-scheduler.readthedocs.io/en/stable/versioning.html).
We always pin a Lightning version ceiling for stability to avoid downstream breakage from Lightning releases that have yet to be fully validated. If you need a newer Lightning version that isn’t yet fully tested, you can always install FTS from main of course:

```bash
uv pip install 'git+https://github.com/speediedan/finetuning-scheduler.git@main[examples]'
```
Thanks again for raising this, feel free to reach out anytime if you have other issues or want to share more about your use case. :rocket:
